[Linux] Linking issue with extlibs when using gmake2 generator

Hi,

When using gmake2 generator for premake, the linking step of NazaraPhysics3D fails because it can't find libNewton. I compared the linking command of both gmake and gmake2, and it seems that the only reason is gmake2 not adding the correct library path in the ```(LINKCMD)```

Here is the gmake2 command line:

```
g++ -o
"bin/x64/DebugDynamic/libNazaraPhysics3D-d.so"
obj/x64/DebugDynamic/NazaraPhysics3D/Collider3D.o
obj/x64/DebugDynamic/NazaraPhysics3D/NewOverload.o
obj/x64/DebugDynamic/NazaraPhysics3D/PhysWorld3D.o
obj/x64/DebugDynamic/NazaraPhysics3D/Physics3D.o
obj/x64/DebugDynamic/NazaraPhysics3D/RigidBody3D.
-L ../../../thirdparty/lib/common
-L ../../../lib
-L ../../../thirdparty/lib/common/x64
-L /usr/lib64 -m64 -shared
-Wl,-soname=libNazaraPhysics3D-d.so
-lNewton
-lNazaraCore-d
/usr/bin/ld : ne peut trouver -lNewton
```

And the gmake output command:

```
g++ -o
"../../../lib/gmake/x64/libNazaraPhysics3D-d.so"
obj/x64/DebugDynamic/NazaraPhysics3D/Collider3D.o
obj/x64/DebugDynamic/NazaraPhysics3D/NewOverload.o
obj/x64/DebugDynamic/NazaraPhysics3D/PhysWorld3D.o
obj/x64/DebugDynamic/NazaraPhysics3D/Physics3D.o
obj/x64/DebugDynamic/NazaraPhysics3D/RigidBody3D.o
-L../../../thirdparty/lib/gmake/x64
-L../../../lib/gmake/x64
-L../../../thirdparty/lib/common
-L../../../lib
-L../../../thirdparty/lib/common/x64
-L/usr/lib64
-m64
-shared
-Wl,-soname=libNazaraPhysics3D-d.so
-lNewton
-lNazaraCore-d
```

if I add the following path, it builds correctly until the NazaraUnitTests

```
-L../../../thirdparty/lib/gmake/x64
-L../../../lib/gmake/x64
```

Then NazaraUnitTests fails because of linking errors too, against the NDK, which seems reasonable as the output dir is ``build/gmake2/tools/bin/x64/DebugDynamic/`` instead of ``lib/gmake2/x64/``.

After updating premake to the latest git version, I got:

```
==== Building chipmunk (debugdynamic_x64) ====
==== Building lua (debugdynamic_x64) ====
==== Building stb_image (debugdynamic_x64) ====
==== Building NazaraCore (debugdynamic_x64) ====
==== Building NazaraAudio (debugdynamic_x64) ====
==== Building NazaraUtility (debugdynamic_x64) ====
==== Building NazaraPlatform (debugdynamic_x64) ====
==== Building NazaraRenderer (debugdynamic_x64) ====
==== Building NazaraGraphics (debugdynamic_x64) ====
==== Building NazaraLua (debugdynamic_x64) ====
==== Building NazaraNetwork (debugdynamic_x64) ====
==== Building NazaraNoise (debugdynamic_x64) ====
==== Building NazaraPhysics2D (debugdynamic_x64) ====
==== Building NazaraPhysics3D (debugdynamic_x64) ====
==== Building PluginAssimp (debugdynamic_x64) ====
==== Building NazaraSDK (debugdynamic_x64) ====
==== Building NazaraSDKServer (debugdynamic_x64) ====
==== Building DemoDopplerEffect (debugdynamic_x64) ====
Linking DemoDopplerEffect
/usr/bin/ld : ne peut trouver -lNazaraPlatform-d
/usr/bin/ld : ne peut trouver -lNazaraUtility-d
collect2: error: ld a retourné le statut de sortie 1
make[1]: *** [DemoDopplerEffect.make:98: ../../../examples/bin/DemoDopplerEffect] Error 1
make: *** [Makefile:234: DemoDopplerEffect] Error 2
```

So it seems to have been half-fixed, only the last issue is still there: libraries are built inside ``build/modules/bin/x64/DebugDynamic/`` directory.

--- a/build/scripts/common.lua
+++ b/build/scripts/common.lua
@@ -1,7 +1,7 @@
 NazaraBuild = {}
 
 -- I wish Premake had a way to know the compiler in advance
-local clangGccActions = "action:" .. table.concat({"codeblocks", "codelite", "gmake", "xcode3", "xcode4"}, " or ")
+local clangGccActions = "action:" .. table.concat({"codeblocks", "codelite", "gmake*", "xcode3", "xcode4"}, " or ")
 
 function NazaraBuild:AddExecutablePath(path)
 	table.insert(self.ExecutableDir, path)
@@ -13,16 +13,16 @@ function NazaraBuild:AddInstallPath(path)
 end
 
 function NazaraBuild:FilterLibDirectory(prefix, func)
-	filter({"action:codeblocks or codelite or gmake", "architecture:x86", "system:Windows"})
+	filter({"action:codeblocks or codelite or gmake*", "architecture:x86", "system:Windows"})
 		func(prefix .. "mingw/x86")
 
-	filter({"action:codeblocks or codelite or gmake", "architecture:x86_64", "system:Windows"})
+	filter({"action:codeblocks or codelite or gmake*", "architecture:x86_64", "system:Windows"})
 		func(prefix .. "mingw/x64")
 
-	filter({"action:codeblocks or codelite or gmake", "architecture:x86", "system:not Windows"})
+	filter({"action:codeblocks or codelite or gmake*", "architecture:x86", "system:not Windows"})
 		func(prefix .. "gmake/x86")
 
-	filter({"action:codeblocks or codelite or gmake", "architecture:x86_64", "system:not Windows"})
+	filter({"action:codeblocks or codelite or gmake*", "architecture:x86_64", "system:not Windows"})
 		func(prefix .. "gmake/x64")
 
 	filter({"action:vs*", "architecture:x86"})

